### PR TITLE
Support long task timer with Timed annotaion 

### LINKED ISSUE #604: TimedAspect support of LongTaskTimer
As of now [TimedAspect](https://github.com/micrometer-metrics/micrometer/blob/0061a4aeba8c59ba018307cfa03435739c973827/micrometer-core/src/main/java/io/micrometer/core/aop/TimedAspect.java) doesn't care about longTask value set to [@Timed](https://github.com/micrometer-metrics/micrometer/blob/master/micrometer-core/src/main/java/io/micrometer/core/annotation/Timed.java).
Do you have any plan to switch to long task timers in case longTask value is set to true ? 

```
    @Around("execution (@io.micrometer.core.annotation.Timed * *.*(..))")
    public Object timedMethod(ProceedingJoinPoint pjp) throws Throwable {
        Method method = ((MethodSignature) pjp.getSignature()).getMethod();
        Timed timed = method.getAnnotation(Timed.class);
        final String metricName = timed.value().isEmpty() ? DEFAULT_METRIC_NAME : timed.value();

        Timer.Sample sample = Timer.start(registry);
        try {
            return pjp.proceed();
        } finally {
            sample.stop(Timer.builder(metricName)
                    .description(timed.description().isEmpty() ? null : timed.description())
                    .tags(timed.extraTags())
                    .tags(tagsBasedOnJoinpoint.apply(pjp))
                    .publishPercentileHistogram(timed.histogram())
                    .publishPercentiles(timed.percentiles().length == 0 ? null : timed.percentiles())
                    .register(registry));
        }
    }
```

### PR DESCRIPTION
**Pull request by nejckorasa**
Support `LongTaskTimer` with `@Timed` annotation.

Closes #604

### CODE REVIEW

File: micrometer-core/src/main/java/io/micrometer/core/aop/TimedAspect.java [hunk 1]
```diff
@@ -87,13 +97,33 @@ public Object timedMethod(ProceedingJoinPoint pjp) throws Throwable {
...
+                .tags(tagsBasedOnJoinPoint.apply(pjp))
+                .register(registry);
+
+        LongTaskTimer.Sample sample = longTaskTimer.start();
+
+        try {
+            return pjp.proceed();
+        } finally {
+            try {
+                longTaskTimer.stop(sample.task());
```
Comment (shakuzen) [inline micrometer-core/src/main/java/io/micrometer/core/aop/TimedAspect.java]: Can `sample.stop()` be used instead here? Then adding the `task()` method would not be needed.

Comment (nejckorasa) [reply micrometer-core/src/main/java/io/micrometer/core/aop/TimedAspect.java]: Yes, that's much cleaner, I've changed it.

**Comment by nejckorasa** [comment]
@shakuzen I've added tests for it. Also wrapped the creation of `LongTaskTimer` so as not to disrupt the application flow in case of exception.

**Comment by nejckorasa** [comment]
> Thank you very much for the contribution. This looks good to me. Do you have interest/time to add tests also? If not, I can do it. Let me know.

I've tested it... Do the added tests and changes look ok?

File: micrometer-core/src/main/java/io/micrometer/core/aop/TimedAspect.java [hunk 2]
```diff
@@ -92,16 +104,46 @@ public Object timedMethod(ProceedingJoinPoint pjp) throws Throwable {
...
-                        .description(timed.description().isEmpty() ? null : timed.description())
-                        .tags(timed.extraTags())
-                        .tags(EXCEPTION_TAG, exceptionClass)
-                        .tags(tagsBasedOnJoinPoint.apply(pjp))
-                        .publishPercentileHistogram(timed.histogram())
-                        .publishPercentiles(timed.percentiles().length == 0 ? null : timed.percentiles())
-                        .register(registry));
+                                    .description(timed.description().isEmpty() ? null : timed.description())
+                                    .tags(timed.extraTags())
+                                    .tags(EXCEPTION_TAG, exceptionClass)
```
Comment (shakuzen) [inline micrometer-core/src/main/java/io/micrometer/core/aop/TimedAspect.java]: It looks like the exception tag isn't applied to the LongTaskTimer. I can add this post-merge, though, unless there was some reason it wasn't included.

**Comment by shakuzen** [comment]
> I've tested it... Do the added tests and changes look ok?

Looks good. Thank you!

File: micrometer-core/src/main/java/io/micrometer/core/aop/TimedAspect.java (hunk 2 above)
Comment (shakuzen) [reply micrometer-core/src/main/java/io/micrometer/core/aop/TimedAspect.java]: Nevermind. I see now why we can't tag the exception class.

**Pull request merged**